packages/snaps-utils/src/caveats.ts

```diff
@@ -43,4 +43,8 @@ export enum SnapCaveatType {
    * Caveat specifying the CAIP-2 chain IDs that a snap can service, currently limited to `endowment:name-lookup`.
    */
   ChainIds = 'chainIds',
+  /**
+   * Caveat specifying the input that a name lookup snap can service, currently limited to `endowment:name-lookup`.
+   */
+  Matchers = 'matchers',
```

Comment:
Since this will be a global caveat in the PermissionController should we name it something like `nameLookupMatchers`?